Feature/landing screen

hakobyaann opened the pull request:
A custom Button for Landing Screen. (In Models Group)

hakobyaann committed:
Landing Screen UI setup

LandingButton custom Button is created to make the code look more readable.

hakobyaann committed:
Update project.pbxproj

nellyAslanyan commented:
Please add screenshot

hakobyaann commented:
![Screenshot 2024-10-23 at 11 02 28](https://github.com/user-attachments/assets/fd8e8bcf-74b4-41f1-b4bf-c2355d579d4c)

nellyAslanyan commented on Birthday/Birthday/Screens/LandingView.swift:
```diff
@@ -0,0 +1,46 @@
+                        textColor: .mainPink,
+                        backgroundColor: .darkRed,
+                        action: { print("Navigation to Register page") },
+                        cornerRadii: [42, 42, 42, 0]
```
Lets have this in property

nellyAslanyan commented:
Check and fix indentation in affected files

hakobyaann committed:
fixed

Indentation in affected files.
Separate property.

nellyAslanyan commented on Birthday/Birthday/Models/LandingButton.swift:
```diff
@@ -0,0 +1,48 @@
+}
+
+#Preview {
+  LandingButton(title: String.Button.register,
```
title should be on new line, ) closing bracket too

nellyAslanyan commented on Birthday/Birthday/Screens/ContentView.swift:
```diff
@@ -12,7 +12,7 @@ import BirthDayAPI
   
     var body: some View {
-      TabBarView()
+        TabBarView()
```
revert  this change

nellyAslanyan commented on Birthday/Birthday/Utils/Extensions/Texts.swift:
```diff
@@ -8,9 +8,11 @@
-  }
-  
+
+    enum Button {
```
indentation here

hakobyaann committed:
fixed

Indentation is fixed.
Unnecessary changes are reverted.

hakobyaann committed:
Merge branch 'develop' into feature/landing-screen

Mekhak11 commented on Birthday/Birthday/Screens/LandingView.swift:
```diff
@@ -0,0 +1,52 @@
+
+struct EnteranceView: View {
+
+  var body: some View {
```
I think no need for creating separate EntranceView and call it in LandingView, You can divide Entrance view into small parts and add those parts in extension of LandingView then use them:)

Mekhak11 commented on Birthday/Birthday/Models/LandingButton.swift:
```diff
@@ -0,0 +1,50 @@
+  var backgroundColor: Color
+  var action: () -> Void
+  var cornerRadii: [CGFloat]
+
```
cornerRadius is better:)

hakobyaann committed:
updated version

The code is divided into smaller parts in the extension of the view and then used.

hakobyaann committed:
Avoiding magic numbers

fixed corner radius values by making a variable for the constant value of corner radius
